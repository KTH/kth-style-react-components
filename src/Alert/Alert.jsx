import React from 'react'

/**
 * @param {string} message - Message to be displayed
 * @param {('success'|'warning'|'danger')} type - Alert type
 */
function Alert({ message, type }) {
  if (!message) {
    return null
  }

  return (
    <div className={`alert alert-${type}`} role="alert">
      {message}
    </div>
  )
}

export default Alert
