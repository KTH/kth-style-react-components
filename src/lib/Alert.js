import React from 'react'
import classNames from 'classnames'

const Alert = props => {
  const classes = classNames(
    'alert',
    props.color === 'success' && 'alert-success',
    props.color === 'danger' && 'alert-danger',
    (props.color === 'info' || !props.color) && 'alert-info' // default value is info
  )

  return (
    <div aria-live="polite" role="alert" className={classes}>
      {props.children}
    </div>
  )
}

export default Alert
