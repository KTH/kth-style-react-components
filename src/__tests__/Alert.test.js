import React from "react";
import renderer from "react-test-renderer";
import Alert from "../lib/Alert"

const TEST_DATA = {
  color: "success",
  string: "This is a test string that should render inside alert component"
}

describe("Alert", () => {
  it ("renders properly", () => {
    const tree = renderer
      .create(<Alert color={TEST_DATA.color}>
        <p>{TEST_DATA.string}</p>
      </Alert>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});