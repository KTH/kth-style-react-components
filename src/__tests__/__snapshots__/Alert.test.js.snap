// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders properly 1`] = `
<div
  aria-live="polite"
  class="alert alert-success"
  role="alert"
>
  <p />
</div>
`;
